internal/kibana/maintenance_window/create.go

```diff
@@ -0,0 +1,50 @@
+package maintenance_window
+
+import (
+	"context"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/kibana_oapi"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+)
+
+func (r *MaintenanceWindowResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
```

Comment:
It could be useful to have a version check here. (Theoretically it should be enough to check this when creating)

Something like this:
https://github.com/elastic/terraform-provider-elasticstack/blob/ec089b4c8dea548f0607b8ba148d6359153e0857/internal/kibana/role.go#L305-L307

(Btw. does this resource also work in serverless? If yes, that might be something to test when adding such a check, to ensure it works)

Reply:
> But serverVersion is part of the other client. Should I use both here, one for version checks and the other for the actual API calls?

I was wrong in this assumption; the version checks are part of the APIClient to which we already have access at this point. I added a version and a serverless check for all 4 operations in this commit:  [d731442](https://github.com/elastic/terraform-provider-elasticstack/pull/1224/commits/d731442ea661eb2fc19488b26b90142f27939665)